copy/copy.go

```diff
@@ -894,6 +1029,44 @@ func (c *copier) copyBlobFromStream(ctx context.Context, srcStream io.Reader, sr
 		inputInfo = srcInfo
 	}
 
+	// Perform image encryption for valid mediatypes if encryptConfig provided
+	var (
+		encryptMediaType string
+		encrypted        bool
+		finalizer        ocicrypt.EncryptLayerFinalizer
+	)
+	if toEncrypt {
+		encryptMediaType, err = manifest.GetEncryptedMediaType(srcInfo.MediaType)
+		if err != nil {
+			return types.BlobInfo{}, errors.Wrapf(err, "Error encrypting layer %s", srcInfo.Digest)
+		}
+
+		if encryptMediaType != "" && c.encryptConfig != nil {
+			var annotations map[string]string
+			if !decrypted {
+				annotations = srcInfo.Annotations
+			}
+			desc := ocispec.Descriptor{
+				MediaType:   srcInfo.MediaType,
+				Digest:      srcInfo.Digest,
+				Size:        srcInfo.Size,
+				Annotations: annotations,
+			}
+
+			s, fin, err := ocicrypt.EncryptLayer(c.encryptConfig, destStream, desc)
+			if err != nil {
+				return types.BlobInfo{}, errors.Wrapf(err, "Image encryption failed for the digest %+v", srcInfo.Digest)
```

Comment:
This error message has a quite different wording. I prefer keeping it consistent.